projects/mod2_fw_102/src/main.c

```diff
@@ -0,0 +1,56 @@
+// Library inclusions:
+#include <stdint.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// Counters Struct
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+// Counter functions:
+void counter_a_func(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;
+  uint8_t counter_a = 1;
+
+  LOG_DEBUG("Counter A: %i\n", storage->counter_a++);
+
+  soft_timer_start(500000, counter_a_func, storage, NULL);
+}
+
+void counter_b_func(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;
+  uint8_t counter_b = 1;
+
+  LOG_DEBUG("Counter B: %i\n", storage->counter_b++);
+
+  soft_timer_start(1000000, counter_b_func, storage, NULL);
+}
+
+// Main Loop:
+
+int main(void) {
+  // Library initializations:
+  interrupt_init();
+  soft_timer_init();
+
+  // Context initialization:
+  Counters storage = { 0 };
+
+  // Initialize timer loop:
+  soft_timer_start(10000,
```

Comment:
It also looks like your values here are off by a factor of 10 (too small), but you can fix this when you use start_millis